ilastik/shell/gui/reportIssueDialog.py

```diff
@@ -0,0 +1,94 @@
+    """
+    log_path = ilastik.ilastik_logging.default_config.get_logfile_path()
+    marker = b"Starting ilastik"
+    text_size_limit = 1048576  # 1 MB
+    file_size = int(os.path.getsize(log_path))
+    with open(log_path, "rb") as log_file:
+        position = max(0, file_size - text_size_limit)
+        log_file.seek(position)
+        log_text = log_file.read()
+    if marker in log_text:
+        return b"".join(log_text.rpartition(marker)[1:]).decode()
+    else:
+        return log_text.decode()
+
+
+class ReportIssueDialog(QDialog):
+    def __init__(self, parent=None, workflow_report_text=""):
+        super().__init__(parent=parent)
+
+        self.setWindowTitle("Report issue")
+        self.setFixedSize(600, 300)
+
+        main_layout = QVBoxLayout()
+
+        # Create and add the instruction label
+        instruction_label = QLabel()
+        instruction_label.setText(
+            "<p>Sorry you ran into a problem! Please copy the information below, and include it when you "
+            "report your issue.</p>"
+            '<p>The best place to get help is the community forum, using the "ilastik" tag '
+            '(<a href="https://forum.image.sc/tag/ilastik">https://forum.image.sc/tag/ilastik</a>).<br>'
+            "Alternatively, you can report the issue on GitHub ("
+            '<a href="https://github.com/ilastik/ilastik/issues/">https://github.com/ilastik/ilastik/issues/</a>'
+            ') or email us at <a href="mailto:team@ilastik.org">team@ilastik.org</a>.</p>'
+        )
+        instruction_label.setOpenExternalLinks(True)
+        instruction_label.setWordWrap(True)
+        main_layout.addWidget(instruction_label)
+
+        report = (
+            f"ilastik version: {ilastik.__version__}\n"
+            f"OS: {platform.platform()}\n"
+            f"{workflow_report_text}"
+            "Log since last startup:\n"
+            f"```\n{_get_log_since_last_startup()}\n```"
+        )
+        self.report_text = QTextEdit()
+        self.report_text.setPlainText(report)
+        self.report_text.setSizePolicy(QSizePolicy.MinimumExpanding, QSizePolicy.Minimum)
+        main_layout.addWidget(self.report_text)
+
+        button_layout = QHBoxLayout()
+        copy_button = QPushButton("Copy to Clipboard")
+        copy_button.clicked.connect(self.copy_to_clipboard)
+        button_layout.addWidget(copy_button)
+        email_button = QPushButton("Open in Email App")
+        email_button.clicked.connect(self.open_in_email_app)
+        button_layout.addWidget(email_button)
+        close_button = QPushButton("Close")
+        close_button.clicked.connect(self.close)
+        button_layout.addWidget(close_button)
+        main_layout.addLayout(button_layout)
+
+        # Set the layout to the dialog
+        self.setLayout(main_layout)
+        self.show()
+
+    def copy_to_clipboard(self):
+        clipboard = QApplication.clipboard()
+        clipboard.setText(self.report_text.toPlainText())
```

Comment:
Wouldn't it make sense to compress the log text first with zip or gz? Should help a lot with the size issue, and scrolling through 1MB of text in an email is (probably) not what you want. I think it can be done with [`QClipboard.setMimeData`](https://doc.qt.io/qt-6/qclipboard.html#setMimeData).

However, a user has to paste the compressed log as an attachment themselves in this case. Although [some mail clients support attachments in `mailto:`](https://news.ycombinator.com/item?id=24196625), this is not universally supported (tried with GMail, didn't work), and it is sketchy security-wise anyway.

Reply:
As Dominik commented above, we're trying to make this easier for the user, and asking them to manually handle email attachments is probably too much to ask.

We could reduce it down to 500KB and add a phrasing asking users to please shorten the log to the relevant parts if they can identify them. But I'd be willing to risk it and just see what our users do with it.

Reply:
Theoretically we could also gzip the log text and base64 it for transmitting via email. But I'm not a fan of this myself, as it could look weird for users.

Another idea is to just filter out lines with loglevel below INFO.